src/OpenTelemetry.Api/Trace/TracerNew.cs

```diff
@@ -0,0 +1,190 @@
+﻿// <copyright file="TracerNew.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Tracer to record distributed tracing information.
+    /// </summary>
+    public class TracerNew
+    {
+        private ActivitySource activitySource;
+
+        internal TracerNew(ActivitySource activitySource)
+        {
+            this.activitySource = activitySource;
+        }
+
+        /// <summary>
+        /// Gets the current span from the context.
+        /// </summary>
+        public TelemetrySpanNew CurrentSpan
+        {
+            get
+            {
+                return new TelemetrySpanNew(Activity.Current);
+            }
+        }
+
+        /// <summary>
+        /// Starts span.
+        /// </summary>
+        /// <param name="name">Span name.</param>
+        /// <param name="kind">Kind.</param>
+        /// <returns>Span instance.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TelemetrySpanNew StartSpan(string name, SpanKind kind = SpanKind.Internal)
+        {
+            // TODO: Open Question - should we have both StartSpan and StartActiveSpan?
```

Comment:
@reyang @CodeBlanch Please review this section specifically.
I am not very sure if I did the right thing here.
`Activity` does not have the ability to be started, but yet not mark it as active. (unless we manipulate `Activity.Current`, but this isn't very clean).
To make this obvious, should we make the shim only expose `StartActiveSpan` methods? Or should we try to follow the spec as is to have `StartSpan` and `StartActiveSpan` ?

Reply:
Tricky. My initial reaction was let's rename to `StartActiveSpan`. But the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#span-creation) right at this moment does say:
>Span creation MUST NOT set the newly created Span as the currently active Span by default, but this functionality MAY be offered additionally as a separate operation.

That is a pretty firm directive! 😮 

It sounds to me like to satisfy the spec we need `Create` and `Start`? We could build up a `SpanNew` in `Create` (detached from `Activity`, which would require some rework of these objects) and then spin up the `Activity` in `Start` using the built-up data? Have a helper that does both?

Reply:
If we do that - won't we still have the the created span/activity as the currently active one, after start is called.?

if we ever do activity.Start(), then it'll become Activity.Current. If there is a strong need to *not* make the started one as the Activity.Current, then we'll need to restore Activity.Current to the previous state.

Reply:
Two thoughts...

* > If we do that - won't we still have the the created span/activity as the currently active one, after start is called.?
   >
   > if we ever do activity.Start(), then it'll become Activity.Current. If there is a strong need to not make the started one as the Activity.Current, then we'll need to restore Activity.Current to the previous state.

   (I'm going to call the two operations `Create` and `Start` but it could be `Start` and `Activate`, or whatever.)

   We could manipulate Activity.Current. What I was saying was we could rework it so it's not a shim, it's more of a builder. You call `CreateSpan` which gives you a TelemetrySpan disconnected from Activity. You can pass in or add tags, events, links, whatever, it just buffers that up. When you call `StartSpan` on that instance it takes all the buffered pieces of data and creates the Activity, which activates it. Basically we are creating two-stages to be compliant with the spec.

   Does it make sense to do that? Not sure. I like the current shims. Is it important to be 100% spec compliant in the case of CreateSpan vs StartSpan? I am totally fine with merging as-is and dealing with this later.

* Should we ask .NET team if they could give us the two-stage flow? The spec does say `MUST`, maybe it should be in the `ActivitySource` API?

Reply:
We can discuss this more in next SIG meeting. I am open to the "builder" pattern, but that may also not make us 100% compliant with spec.